l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -656,13 +656,35 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
     /// @dev IL1AssetHandler interface, used to undo a failed migration of a chain.
     /// @param _chainId the chainId of the chain
     /// @param _assetId the assetId of the chain's STM
-    /// @param _data the data for the recovery
+    /// @param _data the data for the recovery. The structure for this data is:
+    ///              depositAmount || l2Receiver || chainId || stmData || chainData
     function bridgeRecoverFailedTransfer(
         uint256 _chainId,
         bytes32 _assetId,
         address _depositSender,
         bytes calldata _data
-    ) external payable override onlyAssetRouter onlyL1 {}
+    ) external payable override onlyAssetRouter onlyL1 {
+        (, , , bytes memory stmData, bytes memory chainData) = abi.decode(
+            _data,
+            (uint256, address, uint256, bytes, bytes)
+        );
+
+        delete settlementLayer[_chainId];
+
+        IStateTransitionManager(stateTransitionManager[_chainId]).forwardedBridgeClaimFailedBurn({
```

Comment:
we have:


`bridgeBurn -> forwardedBridgeBurn`
`bridgeMint -> forwardedBridgeMint`

maybe we should rename `forwardedBridgeClaimFailedBurn` to `forwardedBridgeRecoverFailedTransfer`